packages/preview-middleware/test/integration/preview/preview.spec.ts

```diff
@@ -0,0 +1,159 @@
+import {
+    test,
+    startServer,
+    teardownServer,
+    copyProject,
+    getDestinationProjectRoot,
+    getPort,
+    expect
+} from '@sap-ux-private/playwright';
+import type { Page } from '@sap-ux-private/playwright';
+import { writeFile } from 'fs/promises';
+import { join } from 'path';
+
+const buildUrl =
+    (port = 3000) =>
+    () =>
+        `http://localhost:${port}`;
+let getUrl: () => string;
+
+const cwd = join(__dirname, '..', '..', 'fixtures', 'simple-app');
+const testCwd = getDestinationProjectRoot(cwd);
+
+/**
+ * This content will overwrite existing `ui5.yaml` file content
+ */
+const getYamlContent = (ui5Version: string) => {
+    return `
+specVersion: '1.0'
+metadata:
+    name: test-project
+type: application
+server:
+    customMiddleware:
+        - name: preview-middleware
+          afterMiddleware: compression
+          configuration:
+              flp:
+                path: /my/custom/path/preview.html
+                libs: true
+                rta:
+                  layer: CUSTOMER_BASE
+              debug: true
+        - name: ui5-proxy-middleware
+          afterMiddleware: preview-middleware
+          configuration:
+              ui5:
+                  - path: /resources
+                    url: https://ui5.sap.com
+                  - path: /test-resources
+                    url: https://ui5.sap.com
+              version: ${ui5Version}
+        - name: sap-fe-mockserver
+          beforeMiddleware: csp
+          configuration:
+              mountPath: /
+              annotations:
+                  - localPath: ./webapp/localService/annotations.xml
+                    urlPath: /sap/opu/odata/IWFND/CATALOGSERVICE;v=2/Annotations*
+              services:
+                  - urlPath: /sap/opu/odata/myservice
+                    metadataPath: ./webapp/localService/metadata.xml
+                    mockdataPath: ./webapp/localService/data
+                    generateMockData: true
+---
+specVersion: "3.0"
+metadata:
+  name: preview-middleware
+kind: extension
+type: server-middleware
+middleware:
+  path: ../../../dist/ui5/middleware.js
+---
+specVersion: "3.0"
+metadata:
+  name: backend-proxy-middleware
+kind: extension
+type: server-middleware
+middleware:
+  path: ../../../../../backend-proxy-middleware/dist/middleware.js
+`;
+};
```

Comment:
Why do you add this? We want to test against the `preview-middleware`. The other middlewares should be installed.Right?

Reply:
yes, adding as project dependencies mean we test agains release and adding like this actually test current main code base. There might not be much difference code wise, but testing current code base detect regression if any.

Reply:
OK, then please remove the `backend-proxy-middleware`. Once this is submitted we can introduce a similar integration test for the `backend-proxy-middleware`.